dds/DCPS/security/SSL/Certificate.cpp

```diff
@@ -435,6 +434,13 @@ void Certificate::load_cert_bytes(const std::string& path)
 
 #else
   std::ifstream in(path.c_str(), std::ios::binary);
+  const std::ifstream::pos_type begin = in.tellg();
```

Comment:
Isn't this guaranteed to be 0 for a file opened (without "append")?
Does it need to check if the stream is open first?

Reply:
I'm not sure about what's guaranteed, and cppreference isn't giving me a warm fuzzy feeling about it either.
I'll add an earlier check, but when in.fail() is true I'm pretty sure those pos_type values come back as -1 and the rest of it is mostly a no-op.